packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,148 @@
+// @ts-ignore
+import * as jwt from 'jsonwebtoken';
+import { utils, providers } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { EthereumDIDRegistry, EthereumDIDRegistry__factory } from '../ethers';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { RoleDefinitionResolverV2__factory } from '@energyweb/credential-governance/ethers/factories/RoleDefinitionResolverV2__factory';
+import { RoleDefinitionResolverV2 } from '@energyweb/credential-governance/ethers/RoleDefinitionResolverV2';
+import { StorageSettings } from './models';
+import { CredentialResolver } from '.';
+
+export class IssuanceVerification {
+  private _provider: providers.Provider;
+  private _resolver: Resolver;
+  private _registrySetting: RegistrySettings;
+  private _didRegistry: EthereumDIDRegistry;
+  private _roleDefResolver: RoleDefinitionResolverV2;
+  private _credentialResolver: CredentialResolver;
+  private ASSERTION_DELEGATE_TYPE =
+    '0x766572694b657900000000000000000000000000000000000000000000000000';
+
+  /**
+   *
+   * @param provider
+   * @param roleDefResolverAddr
+   * @param registrySetting
+   * @param storageSettings
+   */
+  constructor({
+    provider,
+    roleDefResolverAddr,
+    registrySetting,
+    storageSettings,
+  }: {
+    provider: providers.Provider;
+    roleDefResolverAddr: string;
+    registrySetting: RegistrySettings;
+    storageSettings: StorageSettings;
+  }) {
+    this._provider = provider;
+    this._registrySetting = registrySetting;
+    this._roleDefResolver = RoleDefinitionResolverV2__factory.connect(
+      roleDefResolverAddr,
+      this._provider
+    );
+    this._didRegistry = EthereumDIDRegistry__factory.connect(
+      registrySetting.address,
+      this._provider
+    );
+    this._resolver = new Resolver(this._provider, this._registrySetting);
+    this._credentialResolver = new CredentialResolver(storageSettings);
+  }
+
+  /**
+   * Verifies chain of trust for a given holder's credential
+   * @param {string} subjectDID
+   * @param {string} role
+   */
+  async verifyChainOfTrust(subjectDID: string, role: string) {
+    let hasParent = true;
+    while (hasParent) {
+      const subjectDIDDoc = await this._resolver.read(subjectDID);
+      const service = subjectDIDDoc.service;
+      let serviceEndpoint = '';
+      for (const sv of service) {
+        if (utils.namehash(sv.type) === role) {
+          serviceEndpoint = sv.serviceEndpoint;
+          hasParent = false;
+          break;
+        }
+      }
+      const issuerDID = await this.verifyCredential(serviceEndpoint);
```

Comment:
Here's some suggested pseudo code on how I think this could be make more flexible and generic. It probably isn't perfect but let me know what you think.

For instance, we shouldn't have this logic hardcoded to specific verification code. It would be better to pass a `credentialVerifyCallback` so we can update the verification (e.g. to use DIDKit credential verification) without needing to change this code.
```suggestion
  async verifyChainOfTrust(credential: IVerifiableCredential, verifyCredentialProofCallback: (credential: IVerifiableCredential) => Promise<VerificationResult>) {
    let hasParent = true;
    while (hasParent) {
      const role = parseCredentialForRole(credential);
      const roleDefinition = resolveDefinition(role);
      if (roleDefinition.didIssuer) {
        // check if credential.issuer is in roleDefinition
      }
      else {
         const issuerCredentials = getIssuer(issuerDid).accreditations // ideally, getIssuer should be VUI issuer interface
         const correctCredential = findCredential(issuerCredentials, roleDefintion.requiredRole)
         const proofVerificationResult = await verifyCredentialProofCallback(credential);
         // Repeat the process to get authorization of the issuer of the credential...
      }
```

Reply:
Hmm, sounds good. Let me put more thoughts into it.

Reply:
@jrhender I think we can have two separate APIs for this verification. First being the hardcoded way which is the current implementation and the second with the `credentialVerifyCallback` which consumes any external library.

The first/current verification API relies on the the `role` and holder's `did`, this way it interacts with the source of truths( credential storage, DIDDoc) and verifies the chain of trust. 

While the `credentialVerifyCallback` would be a general VC verification approach, which may and mayn't rely on our `role` based credential ecosystem and can verify the same.

Reply:
Ok, we can give it a try to see how it looks 👍   
I'm just looking ahead and I think we're very soon going to want to use DIDKit verification instead of (or at least in addition to) `ProofVerifier` but I'm open to whatever way you think is best to do that.